atlasdb-dbkvs/src/main/java/com/palantir/atlasdb/keyvalue/dbkvs/impl/oracle/OracleDdlTable.java

```diff
@@ -255,40 +258,46 @@ public void compactInternally() {
+                        LoggingArgs.tableRef(tableRef),
+                        "(Enterprise Edition that requires this user to be able to perform DDL operations)."
+                                + " Please change the `enableOracleEnterpriseFeatures` config to false.",
                         e.getMessage());
             } catch (TableMappingNotFoundException e) {
                 throw new RuntimeException(e);
             }
         } else if (config.enableShrinkOnOracleStandardEdition()) {
             Stopwatch timer = Stopwatch.createStarted();
             try {
-                Stopwatch shrinkAndCompactTimer = Stopwatch.createStarted();
-                conns.get().executeUnregisteredQuery(
-                        "ALTER TABLE " + oracleTableNameGetter.getInternalShortTableName(conns, tableRef)
-                                + " SHRINK SPACE COMPACT");
-                log.info("Call to SHRINK SPACE COMPACT on table {} took {} ms.",
-                        tableRef, shrinkAndCompactTimer.elapsed(TimeUnit.MILLISECONDS));
-
-                Stopwatch shrinkTimer = Stopwatch.createStarted();
-                conns.get().executeUnregisteredQuery(
-                        "ALTER TABLE " + oracleTableNameGetter.getInternalShortTableName(conns, tableRef)
-                                + " SHRINK SPACE");
-                log.info("Call to SHRINK SPACE on table {} took {} ms."
-                                + " This implies that locks on the entire table were held for this period.",
-                        tableRef, shrinkTimer.elapsed(TimeUnit.MILLISECONDS));
+                if (inSafeHours) {
+                    Stopwatch shrinkTimer = Stopwatch.createStarted();
+                    conns.get().executeUnregisteredQuery(
+                            "ALTER TABLE " + oracleTableNameGetter.getInternalShortTableName(conns, tableRef)
+                                    + " SHRINK SPACE");
+                    log.info("Call to SHRINK SPACE on table {} took {} ms."
+                                    + " This implies that locks on the entire table were held for this period.",
+                            LoggingArgs.tableRef(tableRef),
+                            SafeArg.of("elapsed time", shrinkTimer.elapsed(TimeUnit.MILLISECONDS)));
+                } else {
+                    Stopwatch shrinkAndCompactTimer = Stopwatch.createStarted();
+                    conns.get().executeUnregisteredQuery(
+                            "ALTER TABLE " + oracleTableNameGetter.getInternalShortTableName(conns, tableRef)
+                                    + " SHRINK SPACE COMPACT");
+                    log.info("Call to SHRINK SPACE COMPACT on table {} took {} ms.",
+                            LoggingArgs.tableRef(tableRef),
+                            SafeArg.of("elapsed time", shrinkAndCompactTimer.elapsed(TimeUnit.MILLISECONDS)));
+                }
```

Comment:
(confirming these are the right way round - SHRINK SPACE COMPACT only takes row level locks, while SHRINK SPACE needs to reset the HWM)